metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/BatchIngestionRunResource.java

```diff
@@ -104,19 +145,60 @@
       }
 
       log.info("finished deleting {} rows", deletedRows.size());
-      response.setAspectsAffected(deletedRows.size() + rowsDeletedFromEntityDeletion);
-      response.setEntitiesAffected(deletedRows.stream().filter(row -> row.isKeyAspect()).count());
-      response.setAspectRowSummaries(
-          new AspectRowSummaryArray(deletedRows.subList(0, Math.min(100, deletedRows.size()))));
-      return response;
+      int aspectsReverted = deletedRows.size() + rowsDeletedFromEntityDeletion;
+
+      final Map<Boolean, List<AspectRowSummary>> aspectsSplitByIsKeyAspects = aspectRowsToDelete.stream()
+              .collect(Collectors.partitioningBy(AspectRowSummary::isKeyAspect));
+
+      final List<AspectRowSummary> keyAspects = aspectsSplitByIsKeyAspects.get(true);
+
+      final long entitiesDeleted = keyAspects.size();
+      final long affectedEntities = deletedRows.stream()
+              .collect(Collectors.groupingBy(AspectRowSummary::getUrn)).keySet().size();
+
+      final AspectRowSummaryArray rowSummaries = new AspectRowSummaryArray(
+              aspectRowsToDelete.subList(0, Math.min(100, aspectRowsToDelete.size())));
+
+      log.info("computing aspects affected by this rollback...");
+      // Compute the aspects that exist referencing the key aspects we are deleting
+      final List<AspectRowSummary> affectedAspectsList = keyAspects.stream()
+              .map((AspectRowSummary urn) -> _systemMetadataService.findByUrn(urn.getUrn(), false))
+              .flatMap(List::stream)
+              .filter(row -> !row.getRunId().equals(runId))
+              .collect(Collectors.toList());
+
+      long affectedAspects = affectedAspectsList.size();
+      long unsafeEntitiesCount = affectedAspectsList.stream()
+              .collect(Collectors.groupingBy(AspectRowSummary::getUrn)).keySet().size();
+
+      final List<AspectInfo> aspectInfos = affectedAspectsList.stream().map(AspectRowSummary::getUrn)
+              .distinct()
+              .map(urn -> {
+                AspectInfo aspectInfo = new AspectInfo();
+                aspectInfo.setUrn(urn);
+                return aspectInfo;
+              })
+              // Return at most 1 million rows
+              .limit(1000000)
```

Comment:
is there a cheap way to make this an ENV variable so we can configure it from outside if needed without having to change code?

Reply:
I can implement something quick and dirty like system.getEnv() but I'm not sure that would be a good idea. Not sure if Rest.li has support for configuration options or if we use them.